src/app/loans/loans-view/loan-account-actions/add-collateral/add-collateral.component.ts

```diff
@@ -0,0 +1,70 @@
+/** Angular Imports. */
+import { Component, OnInit, Input } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+
+/** Custom Services. */
+import { LoansService } from 'app/loans/loans.service';
+
+/**
+ * Add Collateral component.
+ */
+@Component({
+  selector: 'mifosx-add-collateral',
+  templateUrl: './add-collateral.component.html',
+  styleUrls: ['./add-collateral.component.scss']
+})
+export class AddCollateralComponent implements OnInit {
+
+  @Input() dataObject: any;
+
+  /** Collateral form. */
+  collateralForm: FormGroup;
+  /** Loan Id. */
+  loanId: string;
+
+  /**
+   * Retrieve data from `Resolver`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {Router} router Router.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {LoansService} LoansService loans service.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private router: Router,
+              private route: ActivatedRoute,
+              private loanService: LoansService ) { }
+
+  ngOnInit() {
+    this.setCollateralForm();
```

Comment:
Can you please change the name to `createAddCollateralForm()`, as this may create confusion.